js/index.js

```diff
@@ -26,13 +26,85 @@ const skillsSection = document.getElementById('skills');
 const skillsList = skillsSection.querySelector('ul');
 
 // Loop through the skills array and add each skill to the list
-for (let i = 0; i < skills.length; i++) {
-  // Create a new list item element
-  const skill = document.createElement('li');
+for (let skill of skills) {
+ let li = document.createElement('li');
+ li.innerText = skill;
+ skillsList.appendChild(li);
+}
+let messageForm = document.querySelector("[name='leave_message']");
+let messageSection = document.getElementById('messages');
+let messageList = messageSection.querySelector('ul');
+messageSection.hidden = true;
+let idCounter = 0;
+function makeId() {
+	let id = 'entry' + idCounter++;
+	return id;
+}
 
-  // Set the inner text of the list item element to the current skill
-  skill.innerText = skills[i];
+let entryById={};
+messageForm.addEventListener('submit', (event) => {
+	event.preventDefault();
+	let name = event.target.usersName.value;
+	let email = event.target.usersEmail.value;
+	let message = event.target.usersMessage.value;
+
+	console.log('Name:', name);
+	console.log('Email:', email);
+	console.log('Message:', message);
+	const uid = makeId();
+	let newMessage = document.createElement('li');
+
+	newMessage.innerHTML = `<a  href="mailto:${email} ">${name}  </a><span>wrote: ${message} </span>`;
+	newMessage.setAttribute('id', uid);
+
+	entryById[uid] = {usersName: name, usersEmail: email, usersMessage:message};
+	newMessage.appendChild(makeRemoveButton(uid));
+	newMessage.appendChild(makeEditButton(uid));
+
+	messageList.appendChild(newMessage);
+	messageForm.reset();
+	messageSection.style.display = 'block';
+	messageSection.hidden = false;
+});
+
+function makeRemoveButton(uid){
```

Comment:
No need to pass uid to makeRemoveButton and makeEditButton.  In both cases, they get the event list element via parentNode and can then get the uid using getAttribute('id').

Reply:
fixed